### ISSUE TITLE
RATSIGNAL parsing is slightly bugged

### ISSUE THREAD

**Issue by alterNERDtive** [alterNERDtive/VoiceAttack-profiles#68]
Regex refinement needed and completed. Testing for a bit before backporting.

**Comment by alterNERDtive** [comment]
Also “slightly bugged” means:

1. system names >1000 ly from a landmark will include the “ (x,xxx.xx ly from <landmark>)” bit
1. platforms have a leading color identifier, courtesy of having to do right-to-left parsing

Why right to left, you ask? Because landmark systems don’t include the landmark distance part, and I can’t exclude `()` from the system name, they are valid and systems with them exist.

**Comment by alterNERDtive** [comment]
https://github.com/alterNERDtive/VoiceAttack-profiles/blob/adcec42dbda71fa6e7e663480c16324a1ed24ba0/plugins/RatAttack/RatAttack.cs#L19

**Comment by alterNERDtive** [comment]
If you’re feeling particularly adventurous you can replace the `pattern` in `RatAttack.parseRatsignal`’s inline code block with this string:

```
@"RATSIGNAL - CMDR (?<cmdr>.+) - Reported System: (?<system>.+)( \([0-9,\.]+ LY from .*\))? - Platform: \x03[236](?<platform>(PC|XB|PS4))\x03 - O2: (\x034)?(?<oxygen>(NOT )?OK)\x03? Language: .+ \(Case #(?<number>\d+)\) \((PC|XB|PS)_SIGNAL\)"
```

**Comment by alterNERDtive** [comment]
Fixed in 3.1.2, forgot to close this automagically :)
